pkg/web/leaderboard.go

```diff
@@ -1,16 +1,79 @@
 package web
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
 	"net/http"
 	"strconv"
+	"strings"
+	"sync"
 
 	"github.com/nint8835/scribe/pkg/database"
 	"github.com/nint8835/scribe/pkg/web/ui/pages"
 )
 
 const QUOTES_PER_PAGE = 10
 
+// Cache to store input/output pairs
+var cache = make(map[string]string)
+var cacheMutex sync.Mutex // Mutex to handle concurrent access to the cache
+
+// GetGlobalName queries the DiscordLookup API and retrieves the global_name.
+func GetGlobalName(id string) (string, error) {
```

Comment:
A bit hacky, but rather than duplicating all this user handling logic and having a separate cache and everything, could you just generate a string and hand it off to Goldmark, where I've got a plugin that can render discord mentions? You can just generate a string like `<@USER_ID_1>, <@USER_ID_2>` and render it from markdown to HTML and it'll auto-resolve them for you